solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrJdkClient.java

```diff
@@ -0,0 +1,453 @@
+    // knows if Solr can accept Http/2, and no additional
+    // upgrade requests will be sent.
+    //
+    // See https://bugs.openjdk.org/browse/JDK-8287589
+    // See https://github.com/jetty/jetty.project/issues/9998#issuecomment-1614216870
+    //
+    try {
+      ping();
+    } catch (Exception e) {
+      // ignore
+    }
+
+    assert ObjectReleaseTracker.track(this);
+  }
+
+  @Override
+  public NamedList<Object> request(SolrRequest<?> solrRequest, String collection)
+      throws SolrServerException, IOException {
+    checkClosed();
+    if (ClientUtils.shouldApplyDefaultCollection(collection, solrRequest)) {
+      collection = defaultCollection;
+    }
+    String url = getRequestPath(solrRequest, collection);
+    ResponseParser parser = responseParser(solrRequest);
+    ModifiableSolrParams queryParams = initalizeSolrParams(solrRequest);
+    HttpResponse<InputStream> resp = null;
+    try {
+      var reqb = HttpRequest.newBuilder();
+      switch (solrRequest.getMethod()) {
+        case GET:
+          {
+            resp = doGet(url, reqb, solrRequest, queryParams);
+            break;
+          }
+        case POST:
+          {
+            resp = doPutOrPost(url, false, reqb, solrRequest, queryParams);
+            break;
+          }
+        case PUT:
+          {
+            resp = doPutOrPost(url, true, reqb, solrRequest, queryParams);
+            break;
+          }
+        default:
+          {
+            throw new IllegalStateException("Unsupported method: " + solrRequest.getMethod());
+          }
+      }
+      return processErrorsAndResponse(solrRequest, parser, resp, url);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException(e);
+    } catch (HttpTimeoutException e) {
+      throw new SolrServerException(
+          "Timeout occurred while waiting response from server at: " + url, e);
+    } catch (SolrException se) {
+      throw se;
+    } catch (URISyntaxException | RuntimeException re) {
+      throw new SolrServerException(re);
+    } finally {
+      // See
+      // https://docs.oracle.com/en/java/javase/17/docs/api/java.net.http/java/net/http/HttpResponse.BodySubscribers.html#ofInputStream()
+      if (!wantStream(parser)) {
+        try {
+          resp.body().close();
+        } catch (Exception e1) {
+          // ignore
+        }
+      }
+    }
+  }
+
+  private HttpResponse<InputStream> doGet(
+      String url,
+      HttpRequest.Builder reqb,
+      SolrRequest<?> solrRequest,
+      ModifiableSolrParams queryParams)
+      throws IOException, InterruptedException, URISyntaxException {
+    validateGetRequest(solrRequest);
+    reqb.GET();
+    decorateRequest(reqb, solrRequest);
+    reqb.uri(new URI(url + "?" + queryParams));
+    return client.send(reqb.build(), HttpResponse.BodyHandlers.ofInputStream());
+  }
+
+  private HttpResponse<InputStream> doPutOrPost(
+      String url,
+      boolean isPut,
```

Comment:
instead of a boolean, can you use the enum reference?  Boolean args are not very clear.  And this would allow you to combine both case statements back in the switch.

Reply:
Not sure it buys much, but I went ahead and made this change.